lib/client-retry.js

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+"use strict";
+const { AssetComputeClient } = require("./client");
+const colors = require("colors/safe");
+const sleep = require("util").promisify(setTimeout);
+
+const MAX_RETRY_429 = 4;
+const MAX_RETRY_WAIT_TIME = 60000;
+const MIN_RETRY_WAIT_TIME = 30000;
```

Comment:
future improvement: 
- make the backup waittime configurable (if it even would be used)
- remove the backup waittime (if its not ever used)

Reply:
```suggestion
const MIN_RETRY_WAIT_TIME_MS = 30000;
```

Reply:
(unit)